### ISSUE TITLE
Error when inlining private method into nested class

### ISSUE THREAD

**Issue by WalkerCodeRanger** [oleg-st/InlineMethod.Fody#7]
I can get an exception from within `InlineMethod.Fody`. The code compiles and runs fine without the `[Inline]` attribute. I'm using `InlineMethod.Fody` 0.7.1 with `Fody` 6.8.1. It seems to have something to do with the nested class. Code that reproduces it is:

```csharp
using InlineMethod;

internal sealed class Example
{
    private ulong? field;

    [Inline]
    private void ModifyField()
        => field ??= 5;

    public readonly struct Nested
    {
        private readonly Example outer;

        internal Nested(Example outer)
        {
            this.outer = outer;
        }

        public void Caller() => outer.ModifyField();
    }
}
```
The error is:

```plain
Fody: An unhandled exception occurred:
Exception:
Member 'System.Void ModifyField()' is declared in another module and needs to be imported
Type:
System.ArgumentException
StackTrace:
   at Mono.Cecil.MetadataBuilder.LookupToken(IMetadataTokenProvider provider) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 2289
   at Mono.Cecil.Cil.CodeWriter.WriteOperand(Instruction instruction) in C:\projects\fody\cecil\Mono.Cecil.Cil\CodeWriter.cs:line 270
   at Mono.Cecil.Cil.CodeWriter.WriteInstructions() in C:\projects\fody\cecil\Mono.Cecil.Cil\CodeWriter.cs:line 177
   at Mono.Cecil.Cil.CodeWriter.WriteResolvedMethodBody(MethodDefinition method) in C:\projects\fody\cecil\Mono.Cecil.Cil\CodeWriter.cs:line 120
   at Mono.Cecil.Cil.CodeWriter.WriteMethodBody(MethodDefinition method) in C:\projects\fody\cecil\Mono.Cecil.Cil\CodeWriter.cs:line 57
   at Mono.Cecil.MetadataBuilder.AddMethod(MethodDefinition method) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1680
   at Mono.Cecil.MetadataBuilder.AddMethods(TypeDefinition type) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1672
   at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1473
   at Mono.Cecil.MetadataBuilder.AddNestedTypes(TypeDefinition type) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1602
   at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1490
   at Mono.Cecil.MetadataBuilder.AddTypes() in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1444
   at Mono.Cecil.MetadataBuilder.BuildTypes() in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1295
   at Mono.Cecil.MetadataBuilder.BuildModule() in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1066
   at Mono.Cecil.MetadataBuilder.BuildMetadata() in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 1036
   at Mono.Cecil.ModuleWriter.<>c.<BuildMetadata>b__2_0(MetadataBuilder builder, MetadataReader _) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 148
   at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TItem item, Func`3 read) in C:\projects\fody\cecil\Mono.Cecil\ModuleDefinition.cs:line 971
   at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 121
   at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters) in C:\projects\fody\cecil\Mono.Cecil\AssemblyWriter.cs:line 78
   at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters) in C:\projects\fody\cecil\Mono.Cecil\ModuleDefinition.cs:line 1161
   at InnerWeaver.WriteModule() in C:\projects\fody\FodyIsolated\ModuleWriter.cs:line 17
   at InnerWeaver.Execute() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 109
Source:
Mono.Cecil
TargetSite:
Mono.Cecil.MetadataToken LookupToken(Mono.Cecil.IMetadataTokenProvider)
	Compiler.Core		1	
```

**Comment by oleg-st** [comment]
The error occurs because the method was removed but not inlined. The method was removed because of the `[Inline(InlineBehavior.RemovePrivate)]` behavior, which means to remove if the method is private. 
And the method was not inlined because the method is called by the `callvirt` instruction, not the `call`. See https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgH2wDsMYoiBDXAAgGcYqYATagAQCZqBRBCgWwAOuGAFgAUAG9x1GdQFRsANwqlqAV1wQiAcwD81AGbYYuJgG5x02fKUqYbFNQCyEJtgMBPAGLHTACgBKK1lZAF4APkNfFl1dUOoAVgsxYLYAZmpYCiYtXA86DCg1MAxqADkYWlImVKkxEJCbZVUsnKI87l5BYWoINVIoZNSQ4gHKGgqq5j8efiF7PoGg+oaZOtXVjAALbFoAOkWyanjDweHZAF9LFdXWDNZHAGEqYShA48jTvZc3Tx8TJiBZIhK5iUFAA==.

I think in some cases we should handle `callvirt` instructions as well.

**Comment by oleg-st** [comment]
@WalkerCodeRanger Released 0.7.2 with a fix

**Comment by WalkerCodeRanger** [comment]
Awesome, thanks for the rapid response. I've upgraded and verified that 0.7.2 does fix the issue and it is inlined.
